.github/scripts/cmd/cmd.py

```diff
@@ -0,0 +1,204 @@
+                print(f'-- running: {search_manifest_path}')
+                manifest_path = os.popen(search_manifest_path).read()
+                package_dir = os.path.dirname(manifest_path)
+                print(f'-- package_dir: {package_dir}')
+                print(f'-- manifest_path: {manifest_path}')
+                output_path = os.path.join(package_dir, "src", "weights.rs")
+            else:
+                default_path = f"./{config['path']}/src/weights"
+                xcm_path = f"./{config['path']}/src/weights/xcm"
+                output_path = default_path if not pallet.startswith("pallet_xcm_benchmarks") else xcm_path
+            print(f'-- benchmarking {pallet} in {runtime} into {output_path}')
+            cmd = f"frame-omni-bencher v1 benchmark pallet --extrinsic=* --runtime=target/{profile}/wbuild/{config['package']}/{config['package'].replace('-', '_')}.wasm --pallet={pallet} --header={header_path} --output={output_path} --wasm-execution=compiled --steps=50 --repeat=20 --heap-pages=4096"
+            print(f'-- Running: {cmd}')
+            status = os.system(cmd)
+            if status != 0 and not args.continue_on_fail:
+                print(f'Failed to benchmark {pallet} in {runtime}')
+                sys.exit(1)
+
+            # Otherwise collect failed benchmarks and print them at the end
+            # push failed pallets to failed_benchmarks
+            if status != 0:
+                failed_benchmarks[f'{runtime}'] = failed_benchmarks.get(f'{runtime}', []) + [pallet]
+            else:
+                successful_benchmarks[f'{runtime}'] = successful_benchmarks.get(f'{runtime}', []) + [pallet]
+
+    if failed_benchmarks:
+        print('❌ Failed benchmarks of runtimes/pallets:')
+        for runtime, pallets in failed_benchmarks.items():
+            print(f'-- {runtime}: {pallets}')
+
+    if successful_benchmarks:
+        print('✅ Successful benchmarks of runtimes/pallets:')
+        for runtime, pallets in successful_benchmarks.items():
+            print(f'-- {runtime}: {pallets}')
+
+    tempdir.cleanup()
+
+elif args.command == 'fmt':
+    command = f"cargo +nightly fmt"
+    print(f'Formatting with `{command}`')
+    nightly_status = os.system(f'{command}')
+    taplo_status = os.system('taplo format --config .config/taplo.toml')
+
+    if (nightly_status != 0 or taplo_status != 0) and not args.continue_on_fail:
+        print('❌ Failed to format code')
+        sys.exit(1)
+
+elif args.command == 'update-ui':
+    command = '''
+        cargo test --manifest-path substrate/primitives/runtime-interface/Cargo.toml ui
+        cargo test -p sp-api-test ui
+        cargo test -p frame-election-provider-solution-type ui
+        cargo test -p frame-support-test --features=no-metadata-docs,try-runtime,experimental ui
+        cargo test -p xcm-procedural ui
+    '''
+    print(f'Updating ui with `{command}`')
+    status = os.system(f'{command}')
```

Comment:
Were there not also some ENV variables that have to be set to overwrite the UI test cases?

Reply:
you're right, i've missed those. I changed it to use the source of `scripts/update-ui-tests.sh`